pkg/supervisor/supervisor.go

```diff
@@ -134,10 +139,10 @@ func (s *Supervisor) Supervise() error {
 			} else {
 				if s.quit == nil {
 					s.log.Info("Started successfully, go nuts")
-					s.quit = make(chan bool)
-					s.done = make(chan bool)
+					s.quit = make(chan struct{})
```

Comment:
IMO, logic in that file should be refactored to use root context coming from cobra command (or any other caller) and use context.Stop instead of manually orchestrating channels, wdyt?

Reply:
I agree, but I think we can do that in a separate PR.

Reply:
I dropped the "Close channels used by supervisor" commit.

Reply:
sounds fair